File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ds = {}
        dt = {}

        for c in s:
            if c in ds:
                ds[c] += 1
            else :
                ds[c] = 1

        for c in t:
            if c in dt:
                dt[c] += 1
            else :
                dt[c] = 1

        return ds == dt
        

File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {

        uint32_t rev = 0;
        
        for(int i = 0; i < 32; i++){
            int bit = ((n >> i) & 1);
            if(bit == 1){
                rev |= (1 << (31 - i));
            }
        }

        return rev;
    }
};
